hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteFileP.java

```diff
@@ -236,6 +248,33 @@ private void abortUnfinishedTransactions() {
         try (Stream<Path> fileStream = Files.list(directory)) {
             fileStream
                     .filter(file -> file.getFileName().toString().endsWith(TEMP_FILE_SUFFIX))
+                    .filter(file -> {
+                        assert utility.usesTransactionLifecycle();
+                        Matcher m = FILE_INDEX_WITH_SEQ.matcher(file.getFileName().toString());
+                        if (!m.find() || m.groupCount() < 1) {
+                            context.logger().warning("file with unknown name structure found in the directory: " + file);
+                            return false;
+                        }
+                        int index;
+                        try {
+                            index = Integer.parseInt(m.group(1));
+                        } catch (NumberFormatException e) {
+                            context.logger().warning(
+                                    "file with unknown name structure found in the directory: " + file, e);
+                            return false;
+                        }
+                        int localIndexLow = context.memberIndex() * context.localParallelism();
+                        int localIndexHigh = localIndexLow + context.localParallelism();
+                        boolean isLocalIndex = index >= localIndexLow && index < localIndexHigh;
+                        if (sharedFileSystem || isLocalIndex) {
+                            // If the file index belongs to one of the local processors (when directory is not shred)
+                            // or when the directory is shared, we must leave that processor to do the cleanup
```

Comment:
why don't we always leave the cleanup to the processor, looks like this will simplify the logic ?

Reply:
Because there might be no local processor with that index. For example, if LP=1 and you have 2 members. Member 0 will fail and be restarted. After restart it will have index 1, no member on local machine will have index 0. Therefore the files with index 0 will not be handled.

I agree it's quite complex and I was thinking about just ignoring it: the issue would be that some `.tmp` files will be left behind, but that's it.